app/src/main/java/com/devaseemsharma/taskyapp/MainActivity.kt

```diff
@@ -3,32 +3,70 @@ package com.devaseemsharma.taskyapp
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.foundation.isSystemInDarkTheme
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.view.WindowCompat
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.devaseemsharma.taskyapp.authentication.presentation.login.LoginScreen
+import com.devaseemsharma.taskyapp.authentication.presentation.register.RegisterScreen
 import com.devaseemsharma.taskyapp.ui.theme.TaskyAppTheme
+import com.devaseemsharma.taskyapp.utils.NavRoute
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        WindowCompat.setDecorFitsSystemWindows(window, false)
         setContent {
+            val navController = rememberNavController()
             TaskyAppTheme {
-                // A surface container using the 'background' color from the theme
                 Surface(
                     modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
+                    color = MaterialTheme.colorScheme.background,
+
                 ) {
-                    Greeting("Android")
+                    NavGraph(navController = navController)
                 }
             }
         }
     }
 }
 
+
+@Composable
+fun NavGraph(navController: NavHostController) {
+    NavHost(
+        navController = navController,
+        startDestination = NavRoute.Login.path,
+    ) {
+        composable(route = NavRoute.Login.path) {
+            //call LoginScreen composable function here
```

Comment:
These comments don't add any value, I personally only use comments to explain why something happens if that's unclear. Your function, variable and class names should already explain what happens